src/lifecycle/events.js

```diff
@@ -16,7 +25,8 @@ function makeDelegateHandler (elem, handler, parsed) {
     let selector = parsed.selector;
     while (current && current !== elem.parentNode) {
       if (matches(current, selector)) {
-        e.delegateTarget = current;
+        readonly(e, 'currentTarget', current);
+        readonly(e, 'delegateTarget', elem);
```

Comment:
Is there any particular reason that this uses `Object.defineProperty` instead of just `e.delegateTarget = elem`?